runtime/parachains/src/hrmp.rs

```diff
@@ -921,6 +924,19 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// Returns constraints for incoming messages at the given block.
+	pub(crate) fn inbound_hrmp_constraints(
+		recipient: ParaId,
+	) -> InboundHrmpLimitations<T::BlockNumber> {
+		let valid: Vec<T::BlockNumber> = <Self as Store>::HrmpChannelDigests::get(&recipient)
+			.into_iter()
+			.map(|(block_no, _)| block_no)
+			.collect();
+		// Note that inclusion emulator already checks if the watermark is equal
```

Comment:
I would prefer it if this code didn't reference the inclusion emulator, which is strictly further up the stack

Reply:
Comments on the `InboundHrmpLimitations` struct would be better, I think